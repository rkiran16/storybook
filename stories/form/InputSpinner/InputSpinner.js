import { html } from "lit-html";

export const InputSpinner = ({ disabled }) => {

  const isDisabled = disabled? 'disabled' : '';

  return html`
    <div class=${["input-spinner", isDisabled].join(" ")} for="quantity">
      <button ?disabled=${disabled} onclick="" class="btn icon-button left">
        <span class="material-icons">
          remove
        </span>
      </button>
      <input ?disabled=${disabled} class="quantity" id="quantity" value="0" min="0" max="9999" name="quantity" type="number">
      <label for="quantity">Qty</label>
      <button ?disabled=${disabled} onclick="" class="btn icon-button right">
        <span class="material-icons">
          add
        </span>
      </button>
    </div>
  `
}