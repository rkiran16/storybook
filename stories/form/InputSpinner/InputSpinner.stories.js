import { InputSpinner } from "./InputSpinner";

export default {
  title: 'Forms/InputSpinner',
  parameters: {
    zeplinLink: "https://app.zeplin.io/project/62902a463701ffb0e532df2b/screen/62a9fb01c9b9111d8a460042",
  },
  argTypes: {
    disabled: {
      name: "disabled",
      type: { name: 'boolean', required: false},
      description: "Use when the user cannot proceed or until an input is collected",
      defaultValue: false,
      table: {
         type: { summary: 'boolean' },
         defaultValue: { summary: false },
      },
      control: { type: 'boolean'}
    },
  }
};

const InputSpinnerTemplate = args => InputSpinner(args);

export const Default = InputSpinnerTemplate.bind({});