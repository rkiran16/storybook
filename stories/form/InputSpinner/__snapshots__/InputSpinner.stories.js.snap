// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Forms/InputSpinner Default 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "
    <div class=",
    " for=\\"quantity\\">
      <button ?disabled=",
    " onclick=\\"\\" class=\\"btn icon-button left\\">
        <span class=\\"material-icons\\">
          remove
        </span>
      </button>
      <input ?disabled=",
    " class=\\"quantity\\" id=\\"quantity\\" value=\\"0\\" min=\\"0\\" max=\\"9999\\" name=\\"quantity\\" type=\\"number\\">
      <label for=\\"quantity\\">Qty</label>
      <button ?disabled=",
    " onclick=\\"\\" class=\\"btn icon-button right\\">
        <span class=\\"material-icons\\">
          add
        </span>
      </button>
    </div>
  ",
  ],
  "values": Array [
    "input-spinner ",
    false,
    false,
    false,
  ],
}
`;
